### ISSUE TITLE
[REVIEW]: Course Materials for Data Science in Practice

### ISSUE THREAD

**Issue by whedon** [openjournals/jose-reviews#121]
**Submitting author:** @TomDonoghue (<a href="http://orcid.org/0000-0001-5911-0472">Thomas Donoghue</a>)
**Repository:** <a href="https://github.com/DataScienceInPractice/Site" target ="_blank">https://github.com/DataScienceInPractice/Site</a>
**Version:** v1.1
**Editor:** @magsol
**Reviewer:** @KiCh84, @krother
**Archive:** <a href="https://doi.org/10.5281/zenodo.6568091" target="_blank">10.5281/zenodo.6568091</a>

**:warning: JOSE reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSE is currently operating in a "reduced service mode".

## Status

[![status](https://jose.theoj.org/papers/19ab6fe3aec0080e3c8208eaeaf7a670/status.svg)](https://jose.theoj.org/papers/19ab6fe3aec0080e3c8208eaeaf7a670)

Status badge code:

```
HTML: <a href="https://jose.theoj.org/papers/19ab6fe3aec0080e3c8208eaeaf7a670"><img src="https://jose.theoj.org/papers/19ab6fe3aec0080e3c8208eaeaf7a670/status.svg"></a>
Markdown: [![status](https://jose.theoj.org/papers/19ab6fe3aec0080e3c8208eaeaf7a670/status.svg)](https://jose.theoj.org/papers/19ab6fe3aec0080e3c8208eaeaf7a670)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/DataScienceInPractice/Site" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@KiCh84 & @krother, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/jose-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @magsol know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/jose/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/DataScienceInPractice/Site">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [x] **Version:** v1.1
- [x] **Authorship:** Has the submitting author (@TomDonoghue) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [x] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies?
- [x] **Usage:** Does the documentation explain how someone would adopt the module, and include examples of how to use it?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [x] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [x] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [x] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [x] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [x] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Does the paper clearly state the need for this module and who the target audience is?
- [x] **Description:** Does the paper describe the learning materials and sequence?
- [x] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [x] Could someone else teach with this module, given the right expertise?
- [x] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/jose/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/DataScienceInPractice/Site">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [x] **Version:** v1.1
- [x] **Authorship:** Has the submitting author (@TomDonoghue) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [x] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies?
- [x] **Usage:** Does the documentation explain how someone would adopt the module, and include examples of how to use it?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [x] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [x] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [x] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [x] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [x] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Does the paper clearly state the need for this module and who the target audience is?
- [x] **Description:** Does the paper describe the learning materials and sequence?
- [x] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [x] Could someone else teach with this module, given the right expertise?
- [x] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @KiCh84, @krother it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/jose-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/jose-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/jose.00032 is OK
- 10.5281/zenodo.2799972 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.05 s (773.3 files/s, 506834.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Jupyter Notebook                29              0          24233           1333
Markdown                         5            159              0            278
YAML                             2              8             12             47
TeX                              1              1              0             39
make                             1             29             36             35
JSON                             2              0              0              8
-------------------------------------------------------------------------------
SUM:                            40            197          24281           1740
-------------------------------------------------------------------------------


Statistical information for the repository '4a32af834d93a6eba0263b8d' was
gathered on 2021/07/06.
No commited files with the specified extensions were found.
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00121/jose.00121/10.21105.jose.00121.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00121/jose.00121/10.21105.jose.00121.pdf) :page_facing_up: :point_left:

**Comment by krother** [comment]
@KiCh84 I created an issue for the missing release, so that the final paper refers to a defined state of the repository.

**Comment by whedon** [comment]
:wave: @krother, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @KiCh84, please update us on how your review is going (this is an automated reminder).

**Comment by krother** [comment]
There are currently two issue that I found, and I have created GitHub issues for both.

**Comment by TomDonoghue** [comment]
Hey @krother - thanks a ton for the review! The things you brought up make total sense, I will fix them up soon. 

@KiCh84 - have you had a chance to look at the project and do you have any issues / suggestions? If you have a timeline for adding any comments, I might try and group any updates together!

**Comment by KiCh84** [comment]
Hey Tom ! I have been SWAMPED with things ( not only Professional ) , I
think next week I should be able to get into it in more detail . I do
apologize if there is any inconvenience caused by that :/

**Comment by voytek** [comment]
@KiCh84 no, rush since it sounds like a lot's happening (sorry!) but I wanted to see if you had any additional comments or recommendations for us!

**Comment by KiCh84** [comment]
Dear Bradley,

I am reading through the course material, and I just quickly wanted to
check in and ask you about who the target audience is, for this wonderful
course ?
and I apologise heavily for the delay in the responses .

**Comment by voytek** [comment]
No worries! The target is undergraduate students with some Python experience, and some statistics.

We have a complementary paper for that class that outlines the general target—and our motivations—[here](https://www.tandfonline.com/doi/full/10.1080/10691898.2020.1860725).

Thanks for the kind words, and I hope this clarifies!

**Comment by voytek** [comment]
Just checking back in on this @KiCh84!

**Comment by voytek** [comment]
Checking in again, @krother, @KiCh84

**Comment by labarba** [comment]
@whedon check repository

**Comment by whedon** [comment]
Wordcount for `paper.md` is 1030

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.06 s (665.8 files/s, 436407.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Jupyter Notebook                29              0          24233           1333
Markdown                         5            159              0            278
YAML                             2              8             12             47
TeX                              1              1              0             39
make                             1             29             36             35
JSON                             2              0              0              8
-------------------------------------------------------------------------------
SUM:                            40            197          24281           1740
-------------------------------------------------------------------------------


Statistical information for the repository '6efd259c0298602ec2e996f3' was
gathered on 2021/10/01.
No commited files with the specified extensions were found.
```

**Comment by labarba** [comment]
@openjournals/dev — hi folks! I'm a little confused about the stats on this repo. What does it mean when the stats list 24k lines under "comment" for Jupyter Notebook? The repo has a lot of material that is not picked up by the stats tool?

**Comment by arfon** [comment]
> @openjournals/dev — hi folks! I'm a little confused about the stats on this repo. What does it mean when the stats list 24k lines under "comment" for Jupyter Notebook? The repo has a lot of material that is not picked up by the stats tool?

It looks like `cloc` is detecting 29 notebooks which seems about right. As for the actual LOC within, it's entirely possible that this tool doesn't know how to calculate LOC properly (i.e., is getting confused about what is code in a notebook).

Is there other material that you think it should be detecting?

**Comment by labarba** [comment]
> Is there other material that you think it should be detecting?

I don't know! It just threw me off when I looked at the `loc` and saw a small-ish number, then pulled the repo and saw all those Jupyter notebooks with code and all.

**Comment by xuanxu** [comment]
I suspect the tool is counting as 'Comments' every [cell](https://raw.githubusercontent.com/DataScienceInPractice/Site/main/dsip/tutorials/10-Distributions.ipynb) in the notebooks with `cell_type: "markdown"` (as opposed to `cell_type: code`)

**Comment by magsol** [comment]
Hi @KiCh84! 👋 We'd like to get this review wrapped up. Is there anything you need to finish your review?

**Comment by krother** [comment]
@magsol I created a couple of issues in the repository yet. The two related to the checkboxes are: the lack of a version number (not a big deal on my end). The other is that there are two different licenses in the repo and the paper, I believe this needs to be fixed.

**Comment by voytek** [comment]
Hello?

**Comment by voytek** [comment]
Yeah, we can implement those changes pretty quickly. But we were waiting for all reviewers' comments. Should we just proceed?

cc @TomDonoghue

**Comment by TomDonoghue** [comment]
I'll go ahead and come back to this and address everything currently noted (within the next week or so), so that we get as much as possible ready and done here, then it'll just be any last new review items that come up & last checks.

**Comment by magsol** [comment]
Hi folks! Where are we with this review? @krother @KiCh84 there are still unchecked items on both your lists, are those outstanding or have they been addressed yet? @TomDonoghue @voytek how does the feedback so far look?

**Comment by KiCh84** [comment]
Those are outstanding from my end , i really do not know when i will have
the time to review all of it because of the current work situation , seems
to be madness on my side of things for sure :(

**Comment by magsol** [comment]
Hi all! Touching base again.

@KiCh84 has your work situation improved at all? Any sort of timeline to completing the review? Or do we need to look for other reviewers?

@krother It looks like you're -almost- finished with the review; just a couple more items to check off. Are these items you can finish, or do you need feedback from the authors?

@TomDonoghue @voytek Any feedback for the reviewers?

**Comment by KiCh84** [comment]
Hi ! Could you give me till end of Feb ? That would be awesome , if that's
possible!

**Comment by krother** [comment]
@magsol on my side, the two issues https://github.com/DataScienceInPractice/Site/issues/5 and https://github.com/DataScienceInPractice/Site/issues/6 are still open.

**Comment by magsol** [comment]
Thanks for the update, @krother .

@TomDonoghue @voytek Would you have a chance soon to look into those two remaining issues? If we can knock those out, then by that point hopefully we'll have heard back from @KiCh84 .

**Comment by krother** [comment]
@magsol both issues I mentioned above have been resolved. From my side, everything is clear now :sun_with_face:

**Comment by magsol** [comment]
Excellent, thank you so much @krother!

At this point, I think we're just waiting on @KiCh84's completed review.

**Comment by TomDonoghue** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00121/jose.00121/10.21105.jose.00121.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00121/jose.00121/10.21105.jose.00121.pdf) :page_facing_up: :point_left:

**Comment by TomDonoghue** [comment]
Hey all! I have come back around to this and done a round of edits on the project. As well as responding to current reviewer comments, I fixed up a couple things on the site (including fixing up the project section), and also make some minor updates to the paper (re-generated version available above). 

Thank you to @krother for reviewing this work, and for suggesting some helpful comments! I believe I have updated everything related to your review, but please let me know if I missed anything. 

@KiCh84 - is there anything else you need from us to help you finish the review? I see that you've checked off most of the comments - please let us know if you have any comments or suggestions for helping to get the remaining topics checked off. 

@magsol - thanks for keeping track of this, and sorry I was a little slow at getting back to it recently. I believe we have currently done everything on our side, and are just waiting on the final reviewer checks - please let us know if there is anything else we can do to finish this one up!

**Comment by magsol** [comment]
@KiCh84 I hate to bug you on this, but: it looks like the remaining items on your reviewer checklist are pretty straightforward and wouldn't take long for you to verify. Would you mind taking a look sometime this week to see if everything checks out? Thank you so much!!

**Comment by TomDonoghue** [comment]
Just wanted to check back in here, since I think we're hopefully quite close, and it would be really nice to finish this. 

@KiCh84 - do you know if / when you'll be able to finish the review here?
@magsol - in the mean time, is there anything else we can do to help get this finished up?

**Comment by magsol** [comment]
Hi @TomDonoghue , thanks for checking in on things.

@KiCh84: we're waiting for you at this point! If you can't finish in the next week ish, please let me know, as we'd really like to get this over the finish line ASAP.

**Comment by voytek** [comment]
Anything next here?

**Comment by magsol** [comment]
@KiCh84 Hi there 👋 We're waiting on your review to wrap things up here! Please let me know ASAP with either a timetable for wrapping this up, or let me know if we need to find a replacement reviewer. Thank you!

**Comment by KiCh84** [comment]
Hi Shannon , It looks you might have to find a replacement reviewer .
There's just too many things on my plate currently and would be unable to
give it the attention it deserves . I apologize heavily for this :(

**Comment by TomDonoghue** [comment]
Okay... @magsol - can you clarify what happens now and what the expected timeline is? It's already a year since this started review, and it was a year before that that we actually submitted this project...

**Comment by magsol** [comment]
@TomDonoghue At this point, I think it'd be easiest for me to go through the outstanding points on @KiCh84's reviewer checklist. Fortunately, most of the items are finished; the unresolved ones seem to revolve around the paper itself. I should have some time tomorrow to go through them and then, providing everything checks out, we can start the publication process.

I sincerely apologize for how long it's taken. Let's see if we can't get this wrapped by the weekend.

**Comment by TomDonoghue** [comment]
@magsol - thank you, that sounds like a practical and efficient way to try and wrap this up! Please let us know of any comments and suggestions for the paper / project, and I will try to address them as soon as I can!

**Comment by magsol** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00121/jose.00121/10.21105.jose.00121.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00121/jose.00121/10.21105.jose.00121.pdf) :page_facing_up: :point_left:

**Comment by magsol** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/jose.00032 is OK
- 10.5281/zenodo.4539666 is OK
- 10.1080/10691898.2020.1860725 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by magsol** [comment]
@TomDonoghue Hi Tom, I'm bringing things over the finish line (they all look great!), and the last thing I need from you: the archive DOI. Once you've given me that I'll wrap things up.

**Comment by TomDonoghue** [comment]
Thanks @magsol - for the quick checks here!

I have added Zenodo to the org / repo in order to create an archive DOI:
Here's the Zenodo link: https://zenodo.org/record/6568091
Here's the DOI: 10.5281/zenodo.6568091

Note: I had to create a new release to start Zenodo tracking (or, at least, I wasn't sure how to retro-actively add a Zenodo DOI to a previous release), so there is now a 1.1 release to kickstart this, though there are no content changes to what is tagged under the 1.0 release (see release page: https://github.com/DataScienceInPractice/Site/releases). I don't think this matters in any meaningful way, just wanted to note it in case it's important to update a version number to 1.1 somewhere, since the project was previously tagged to 1.0.

**Comment by magsol** [comment]
@whedon set 10.5281/zenodo.6568091 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.6568091" target="_blank">10.5281/zenodo.6568091</a> is the archive.

**Comment by magsol** [comment]
@whedon set v1.1 as version

**Comment by whedon** [comment]
OK. v1.1 is the version.

**Comment by magsol** [comment]
(@TomDonoghue , thanks for the heads-up on the version change!)

**Comment by magsol** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/jose.00032 is OK
- 10.5281/zenodo.4539666 is OK
- 10.1080/10691898.2020.1860725 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/jose-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/jose-papers/pull/97

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/jose-papers/pull/97, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by labarba** [comment]
We request that authors edit the metadata of the Zenodo deposit so title and author list match the JOSE paper. It's just cleaner that way as readers see these as part of the "same scholarly object." Could you do that? (Note that you may not want Zenodo to do _automatic_ updates of versions with each release. Note also that Zenodo pulls every committer into the author list.)

**Comment by TomDonoghue** [comment]
I've updated the Zenodo record to match the title of the JOSE paper, and checked that the authors match. 
This should now be updated: https://zenodo.org/record/6568091

**Comment by labarba** [comment]
- [ ] for the JupyterBooks reference, please add `{}` around the author `Executable Books Community` so that it is rendered like that, and not like "Community, E. B."

**Comment by labarba** [comment]
- [ ] on [line 59](https://github.com/DataScienceInPractice/Site/blob/8b7dc7ce4bda75bf234363b69f4b7c6be30acf6e/paper/paper.md?plain=1#L59): "This set of materials were" >> was

**Comment by TomDonoghue** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00121/jose.00121/10.21105.jose.00121.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00121/jose.00121/10.21105.jose.00121.pdf) :page_facing_up: :point_left:

**Comment by TomDonoghue** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00121/jose.00121/10.21105.jose.00121.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00121/jose.00121/10.21105.jose.00121.pdf) :page_facing_up: :point_left:

**Comment by TomDonoghue** [comment]
@labarba - thanks for the detailed check, I have now fixed the reference and wording issues you noticed! Let me know if there is anything else that needs checking / fixing!

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSE!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/jose-papers/pull/98
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/jose.00121](https://doi.org/10.21105/jose.00121)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by labarba** [comment]
Congratulations, @TomDonoghue, your JOSE paper is published! 🚀 

Huge thanks to our Editor: @magsol, and the Reviewers: @KiCh84, @krother — your contributions make this adventure in scholarly publishing possible 🙏

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://jose.theoj.org/papers/10.21105/jose.00121/status.svg)](https://doi.org/10.21105/jose.00121)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/jose.00121">
  <img src="https://jose.theoj.org/papers/10.21105/jose.00121/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://jose.theoj.org/papers/10.21105/jose.00121/status.svg
   :target: https://doi.org/10.21105/jose.00121
```
This is how it will look in your documentation:

[![DOI](https://jose.theoj.org/papers/10.21105/jose.00121/status.svg)](https://doi.org/10.21105/jose.00121)

**We need your help!**

Journal of Open Source Education is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future.
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-jose

**Comment by voytek** [comment]
🎉🎉🎉

**Comment by atulkjhyt456** [comment]
When studying data science in practice, it's essential to have a comprehensive set of course materials to enhance your learning experience. Here are some common materials that can be included in a data science course:

Textbooks and Reference Books: Data science textbooks provide a structured approach to learning the fundamental concepts and techniques in the field. Some popular books include "Python for Data Analysis" by Wes McKinney, "Data Science for Business" by Foster Provost and Tom Fawcett, and "Hands-On Machine Learning with Scikit-Learn, Keras, and TensorFlow" by Aurélien Géron.

Lecture Slides and Presentations: Lecture slides serve as a guide during classroom or online sessions. They typically include key concepts, explanations, examples, and illustrations that support the course material. These slides are often made available for students to review and reinforce their understanding.

Case Studies and Practical Examples: Real-world case studies and practical examples help students apply data science concepts to solve problems. These materials showcase how data science techniques can be used in various industries, such as finance, healthcare, marketing, and e-commerce.

Code Examples and Jupyter Notebooks: Code examples and Jupyter Notebooks provide hands-on experience with programming languages commonly used in data science, such as Python or R. These materials demonstrate how to implement algorithms, preprocess data, build models, and perform data visualizations.

Datasets: Access to curated datasets is crucial for practicing data science techniques. Datasets can include structured data, such as CSV files, as well as unstructured data, such as text documents or images. Datasets should cover a variety of domains to expose students to different data types and challenges.

Assignments and Exercises: Assignments and exercises allow students to apply their knowledge and reinforce concepts covered in the course. These can include coding tasks, data analysis projects, or model-building exercises that require students to practice using data science techniques.

Online Resources and Tutorials: Supplementary online resources, tutorials, and video lectures can provide additional explanations, alternative perspectives, and practical tips. Platforms like DataCamp, Coursera, and Kaggle offer a wide range of data science courses and tutorials that can complement the course materials.

It's worth noting that the specific materials may vary depending on the institution, course structure, and the instructor's preferences. It's always beneficial to have a mix of theoretical content, practical applications, and real-world examples to ensure a well-rounded understanding of data science in practice.

[Data Science Classes in Pune](https://www.sevenmentor.com/data-science-course-in-pune.php)

[Data Science Course in Pune](https://www.sevenmentor.com/data-science-course-in-pune.php)

[Data Science Training in Pune](https://www.sevenmentor.com/data-science-course-in-pune.php)

**Comment by labarba** [comment]
:loudspeaker:  New Reviewer Sign-up Form!
https://forms.gle/Rizd3TcHnQKhrbYY7
